fieldpath/set.go

```diff
@@ -128,6 +128,31 @@ func (s *Set) Has(p Path) bool {
 	}
 }
 
+// HasPartial returns true if PathElement of p is a member of the set.
+// Example:
+// For a set containing .obj
+// any path starting with .obj (like .obj.a) is part of the set.
+func (s *Set) HasPartial(p Path) bool {
```

Comment:
I'm not completely sure what this does nor why it's needed yet. But I'm a little curious.

Reply:
The current Has behaves like this:

set = [.obj]
set.Has(".obj") -> true
set.Has(".obj.a") -> false

HasPartial behaves like this:

set = [.obj]
set.Has(".obj") -> true
set.Has(".obj.a") -> true

It does what the new Remove does, match children if the parent path is defined.

Reply:
OK, If it's not has, there is already a method that does that.

Reply:
`WithPrefix` I think is somewhat similar to what you're trying to do

Reply:
Yeah, i tried around a lot with WithPrefix and didn't get it to work.
It never resulted in a positive match.
Only the Set has Prefix not the Path and prefixing the set didn't help. Also because the merge code does not expose at which step/path it is.
So i tried adding that, but without success.

This is the only way i could get it to work so far :-/